From bf78bafa9586d847996d6d8e426d765026641645 Mon Sep 17 00:00:00 2001
From: Henri Yandell <bayard@apache.org>
Date: Mon, 4 Dec 2006 23:35:54 +0000
Subject: [PATCH] Applied the fix suggested by Mirko Friedenhagen in #IO-100.
 This isn't something that it's easy to write a unit test for, but it is very
 easy to write a platform dependent test and show that the new code correctly
 throws an exception for '/etc/passwd'

git-svn-id: https://svn.apache.org/repos/asf/jakarta/commons/proper/io/trunk@482411 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/commons/io/FileUtils.java | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/java/org/apache/commons/io/FileUtils.java b/src/java/org/apache/commons/io/FileUtils.java
index d8e34a7f38d..2d96b44658b 100644
--- a/src/java/org/apache/commons/io/FileUtils.java
+++ b/src/java/org/apache/commons/io/FileUtils.java
@@ -142,7 +142,10 @@ public static void touch(File file) throws IOException {
             OutputStream out = new FileOutputStream(file);
             IOUtils.closeQuietly(out);
         }
-        file.setLastModified(System.currentTimeMillis());
+        boolean success = file.setLastModified(System.currentTimeMillis());
+        if(!success) {
+            throw new IOException("Unable to set the last modification time for " + file);
+        }
     }
 
     //-----------------------------------------------------------------------
